src/nat/eval/dataset_handler/dataset_filter.py

```
@@ -33,6 +36,32 @@ def __init__(self, filter_config: EvalFilterConfig):
+            Boolean Series indicating matches
+        """
+        # Convert series to string for pattern matching
+        str_series = series.astype(str)
+
+        # Initialize boolean mask
+        matches = pd.Series([False] * len(series), index=series.index)
+
+        # Check each pattern using fnmatch
+        for pattern in patterns:
+            pattern_str = str(pattern)
+            pattern_matches = str_series.apply(lambda x: fnmatch.fnmatch(x, pattern_str))
```

Ruff (0.14.3)

Function definition does not bind loop variable `pattern_str`

(B023)